packages/contentful--create-contentful-app/src/getTemplateSource.ts

```diff
@@ -0,0 +1,108 @@
+
+import fetch from 'node-fetch';
+import inquirer from 'inquirer';
+import chalk from 'chalk';
+import { track } from './analytics';
+import { CLIOptions, ContentfulExample } from './types';
+import { highlight, warn, wrapInBlanks } from './logger';
+
+const EXAMPLES_PATH = 'contentful/apps/examples/';
+
+function isContentfulTemplate(url: string) {
+  return Object.values(ContentfulExample).some((t) => url.includes(EXAMPLES_PATH + t));
+}
+
+async function getGithubFolderNames(username: string, repo: string, path: string) {
+  const url = `https://api.github.com/repos/${username}/${repo}/contents/${path}`;
+
+  const response = await fetch(url);
+  const contents: any = await response.json();
+
+  return contents
+    .filter((content: any) => content.type === 'dir')
+    .map((content: any) => content.name);
+}
+
+
+
+async function promptExampleSelection(): Promise<string> {
+  let template = 'typescript';
+
+  // ask user whether to start with a blank template or use an example template
+  const { starter } = await inquirer.prompt([
+    {
+      name: 'starter',
+      message: 'Do you want to start with a blank template or use one of our examples?',
+      type: 'list',
+      choices: ['blank', 'example'],
+      default: 'blank',
+    },
+  ]);
+
+  // if the user chose to use an blank template, ask which language they prefer
+  if (starter === 'blank') {
+    const { language } = await inquirer.prompt([
+      {
+        name: 'language',
+        message: 'Do you prefer TypeScript or JavaScript',
+        type: 'list',
+        choices: [
+          { name: 'TypeScript', value: 'typescript' },
+          { name: 'JavaScript', value: 'javascript' },
+        ],
+        default: 'typescript',
+      },
+    ]);
+    template = language;
+  } else {
+    // get available templates from examples
+    const availableTemplates = await getGithubFolderNames('contentful', 'apps', 'examples');
+
+    // ask user to select a template from the available examples
+    const { example } = await inquirer.prompt([
+      {
+        name: 'example',
+        message: 'Select a template',
+        type: 'list',
+        choices: availableTemplates,
+      },
+    ]);
+
+    template = example;
+  }
+
+  // return the selected template
+  return selectTemplate(template);
+}
+
+function selectTemplate(template: string) {
+  wrapInBlanks(highlight(`---- Cloning the ${chalk.cyan(template)} template...`));
+  return EXAMPLES_PATH + template;
+}
+
+
+async function makeContentfulExampleSource(options: CLIOptions): Promise<string> {
+  if (options.example) {
```

Comment:
Maybe refactor all this to `switch`?

Reply:
This code was just copy-pasted, but I can change it 👍

Reply:
actually, we are checking here for the existence of different object properties, don't think a switch statement would make much sense here 🤔 Or how would you do that?

Reply:
Ah, yeah, sorry, my bad.